tests/31sync/06state.pl

```diff
@@ -1,4 +1,30 @@
-test "State is included in the initial sync",
+use Future::Utils qw( repeat );
+
+# call /sync repeatedly until it returns a result
+# with an event in the given room
+sub wait_for_event_in_room {
```

Comment:
Well, I could, but I specifically want to use ``/sync`` rather than ``/events``, so I'd have to have an extra call to ``/sync`` once I decided there was something worth syncing for.

That's not to say that this shouldn't be implemented in a more generic way...